systemd-257 update patches

vigh-m opened the pull request:
<!--
Tips:
- Please read CONTRIBUTING.md to understand our process and our requests for PRs.
- Please file an issue before creating a PR so we can discuss the change and confirm it's not already being worked on.
-->

**Issue number:**

Related #536

**Description of changes:**

Follow on work from #581 

1. Remove `9001-core-add-separate-timeout-for-system-shutdown.patch` 
    - Tested this by launching nodes with and without the patch and rebooting them while watching running pods on the nodes. Didn't observe any change in the time taken for pods to come back online. 
    - No change in journalctl to indicate extra time taken to shutdown/reboot from when the command was sent to when the shutdown was completed
3. Migrate `9008-core-mount-increase-mount-rate-limit-burst-to-25.patch` to use bootconfigs
    - Added more debug statements to validate the need for the patch and the bootconfig:
    - When `mount_rate_limit_burst` is at the default of 5
    ```
    Aug 18 20:40:47 localhost systemd[1]: Mount rate limit burst configured to 5
    Aug 18 20:40:47 localhost systemd[1]: Starting Generate network units from Kernel command line...
    Aug 18 20:40:47 localhost systemd[1]: Finished Generate network units from Kernel command line.
    Aug 18 20:40:47 localhost systemd[1]: Event source 0x55d1939d1ab0 (mount-monitor-dispatch) entered rate limit state.
    Aug 18 20:40:48 localhost systemd[1]: Mount rate limit expired: 5 events exceeded in 1000000μs interval
    Aug 18 20:40:49 localhost systemd[1]: Event source 0x55d1939d1ab0 (mount-monitor-dispatch) entered rate limit state.
    Aug 18 20:40:50 localhost systemd[1]: Mount rate limit expired: 5 events exceeded in 1000000μs interval
    ```
    - When `mount_rate_limit_burst` is at the updated value of 25 from the patch
    ```
    Aug 18 21:13:07 localhost systemd[1]: Mount rate limit burst configured to 25
    Aug 18 21:13:07 localhost systemd[1]: Starting Generate network units from Kernel command line...
    Aug 18 21:13:07 localhost systemd[1]: Finished Generate network units from Kernel command line.
    Aug 18 21:13:10 localhost systemd[1]: Starting Generate network configuration...
    Aug 18 21:13:10 localhost systemd[1]: Finished Generate network configuration.
    Aug 18 21:13:13 localhost systemd[1]: Starting Generate additional settings for Kubernetes...
    Aug 18 21:13:13 localhost systemd[1]: Finished Generate additional settings for Kubernetes.
    ```
    - With bootconfig:
    ```
    bash-5.1# journalctl | grep rate
    Aug 19 20:29:06 localhost kernel: Loaded X.509 cert 'Build time autogenerated kernel key: 26d914161d51d66aeaa0d25b16207a74765f29de'
    Aug 19 20:29:06 localhost systemd[1]: Mount rate limit burst configured to 25
    Aug 19 20:29:06 localhost systemd[1]: Starting Generate network units from Kernel command line...
    Aug 19 20:29:06 localhost systemd[1]: Finished Generate network units from Kernel command line.
    Aug 19 20:29:10 localhost systemd[1]: Starting Generate network configuration...
    Aug 19 20:29:10 localhost systemd[1]: Finished Generate network configuration.
    Aug 19 20:29:14 localhost systemd[1]: Starting Generate additional settings for Kubernetes...
    Aug 19 20:29:15 localhost systemd[1]: Finished Generate additional settings for Kubernetes.
    ```

    - The env variable can be seen in `dmesg`, being passed in to userspace
    ```
    [    0.022874] Kernel command line: initcall_blacklist="vmd_drv_init" initcall_blacklist="megasas_init" initcall_blacklist="mpt3sas_init" initcall_blacklist="pqi_init" module_blacklist="i8042" fips="1" SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST="25" BOOT_IMAGE=(hd0,gpt3)/vmlinuz console=tty0 console=ttyS0,115200n8 net.ifnames=0 netdog.default-interface=eth0:dhcp4,dhcp6? quiet bootconfig root=/dev/dm-0 rootwait ro raid=noautodetect random.trust_cpu=on selinux=1 enforcing=1 "dm-mod.create=root,,,ro,0 1884160 verity 1 PARTUUID=4604128e-9d7a-4297-a525-5aa2959874b6/PARTNROFF=1 PARTUUID=4604128e-9d7a-4297-a525-5aa2959874b6/PARTNROFF=2 4096 4096 235520 1 sha256 d8c0ba283f9a6cacb6dcee9d24fafde0a33c436340488ef29135b72d8fd0931c b00b529d82c9b07bef1cb3f2794925d02eb67a617f027c38bd127981e27f7da7 2 restart_on_corruption ignore_zero_blocks" -- systemd.unit="fipscheck.target"  systemd.log_target=journal-or-kmsg systemd.log_color=0 systemd.show_status=true
    [    0.022890] blacklisting initcall vmd_drv_init
    [    0.022899] blacklisting initcall megasas_init
    [    0.022907] blacklisting initcall mpt3sas_init
    [    0.022915] blacklisting initcall pqi_init
    [    0.022932] fips mode: enabled
    [    0.023090] Unknown kernel command line parameters "SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST=25 BOOT_IMAGE=(hd0,gpt3)/vmlinuz selinux=1 enforcing=1", will be passed to user space.
                                                here-------^
    ```

3. Add a patch to disable the kernel command line requirement for cgroupsv1 
    - Tested via kernel cmdline
    ```
    bash-5.1# findmnt /sys/fs/cgroup
    TARGET         SOURCE FSTYPE OPTIONS
    /sys/fs/cgroup tmpfs  tmpfs  ro,nosuid,nodev,noexec,seclabel,size=4096k,nr_inodes=1024,mode=755
    ```
    - `dmesg` output
    ```
    bash-5.1# dmesg | grep unified
    [    0.000000] Command line: BOOT_IMAGE=(hd0,gpt3)/vmlinuz console=tty0 console=ttyS0,115200n8 net.ifnames=0 netdog.default-interface=eth0:dhcp4,dhcp6? quiet systemd.unified_cgroup_hierarchy=0 bootconfig root=/dev/dm-0 rootwait ro raid=noautodetect random.trust_cpu=on selinux=1 enforcing=1 "dm-mod.create=root,,,ro,0 624096 verity 1 PARTUUID=7243be01-2731-4022-948e-d1a5066e19bf/PARTNROFF=1 PARTUUID=7243be01-2731-4022-948e-d1a5066e19bf/PARTNROFF=2 4096 4096 78012 1 sha256 1e9ccc78a6204daeb9c72c4a97da15c52a8adf2725c139ed6a41c5cba4ee6ee4 547b141ab415e53211b42afdddb686f45a0139ac346c72cfa9c9c4b3955a593f 2 restart_on_corruption ignore_zero_blocks" -- systemd.log_target=journal-or-kmsg systemd.log_color=0 systemd.show_status=true
    [    0.015615] Kernel command line: module_blacklist=i8042 SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST=25 BOOT_IMAGE=(hd0,gpt3)/vmlinuz console=tty0 console=ttyS0,115200n8 net.ifnames=0 netdog.default-interface=eth0:dhcp4,dhcp6? quiet systemd.unified_cgroup_hierarchy=0 bootconfig root=/dev/dm-0 rootwait ro raid=noautodetect random.trust_cpu=on selinux=1 enforcing=1 "dm-mod.create=root,,,ro,0 624096 verity 1 PARTUUID=7243be01-2731-4022-948e-d1a5066e19bf/PARTNROFF=1 PARTUUID=7243be01-2731-4022-948e-d1a5066e19bf/PARTNROFF=2 4096 4096 78012 1 sha256 1e9ccc78a6204daeb9c72c4a97da15c52a8adf2725c139ed6a41c5cba4ee6ee4 547b141ab415e53211b42afdddb686f45a0139ac346c72cfa9c9c4b3955a593f 2 restart_on_corruption ignore_zero_blocks" -- systemd.log_target=journal-or-kmsg systemd.log_color=0 systemd.show_status=true
   ```

    - Test via user-data settings:
    ```
    [settings.boot.kernel-parameters]
    "systemd.unified_cgroup_hierarchy"= ["0"]
    
    [settings.boot]
    reboot-to-reconcile = true
    ```
    ```
    bash-5.1# findmnt /sys/fs/cgroup/
    TARGET         SOURCE FSTYPE OPTIONS
    /sys/fs/cgroup tmpfs  tmpfs  ro,nosuid,nodev,noexec,seclabel,size=4096k,nr_inodes=1024,mode=755
    ```



**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

vigh-m committed:
systemd: remove shutdown timeout patch

Testing showed that we do not hit the timeout overriden by Patch9001.
Shutdown is consistent and is not blocked for 30s without
this patch.

Signed-off-by: Vighnesh Maheshwari <vighmah@amazon.com>

arnaldo2792 commented on packages/systemd-257/9001-core-add-separate-timeout-for-system-shutdown.patch:
```diff
@@ -1,84 +0,0 @@
-From e66920d0a95a3e53b24a2f270420a85d0f2395ea Mon Sep 17 00:00:00 2001
```
For the lack of a better place:

nit: in the commit message you are explaining what happened. I can see the patch was removed, but why?

vigh-m replied:
This was a follow up from [this](https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/581#discussion_r2274300182) comment. I can reword the commit message to make it more definitive that this patch is not required since we don't hit either timeout during shutdown

arnaldo2792 commented on packages/release/release.spec:
```diff
@@ -119,6 +119,7 @@ Source1400: logdog.common.conf
 
```
For the lack of a better place to put this:

The commit message is a bit confusing. You actually removed the patch. The part that is confusing is:

> Using bootconfig to set this variable since we don't have a shell environment

You don't need a shell environment to set an ENV variable. And according to the comment that was removed in the spec:

```bash
# Local patch to adjust the default mount rate limit to 25 per second.
# Carried as a patch so that SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST can be used
# as a kernel command line parameter to override.
```

We could set it as a kernel command line. So my question is, why as a boot config?

vigh-m replied:
Setting it at the kernel command line was considered but the bootconfig method was chosen as a way to set defaults for Bottlerocket without requiring every variant definition needing an update

arnaldo2792 commented on packages/systemd-257/9011-remove-cgroup-kernel-cmdline-requirement.patch:
```diff
@@ -0,0 +1,23 @@
+From 6eec85a7673af42ae54c1cd1d0133af68279e1ac Mon Sep 17 00:00:00 2001
+From: Vighnesh Maheshwari <vighmah@amazon.com>
+Date: Fri, 22 Aug 2025 14:03:33 -0700
+Subject: [PATCH] remove-cgroup-kernel-cmdline-requirement
```
You are missing a `Sign-Off-By` here.

arnaldo2792 commented on packages/systemd-257/9011-remove-cgroup-kernel-cmdline-requirement.patch:
```diff
@@ -0,0 +1,23 @@
+index 24d2ebdaf5..07ca308b3f 100644
+--- a/src/shared/cgroup-setup.c
++++ b/src/shared/cgroup-setup.c
+@@ -169,10 +169,6 @@ bool cg_is_legacy_force_enabled(void) {
```
It seems like this function was introduced in:

https://github.com/systemd/systemd/pull/31442/commits/f2512de82dc91cfb742a4f4df934bdb4fcad482d

to forbid exactly what you want. I think you could remove the function call, instead of removing the one check, as the function call is what actually does a bunch of checks to make it harder to enable cgroupsv1.

vigh-m replied:
We're moving to using bootconfigs instead of carrying the patch.  That should be a much cleaner diff

bcressey commented on packages/release/systemd-mount-rate-bootconfig.conf:
```diff
@@ -0,0 +1 @@
+kernel.SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST=25
```
This should be an `init` parameter instead:
```
init.SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST=25
```

(You can verify it's set by checking `/proc/1/environ` after boot.)

vigh-m replied:
`init.SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST=25` doesn't seem to work here.

```
# dmesg output
[    0.309181] Run /sbin/init as init process
[    0.309183]   with arguments:
[    0.309185]     /sbin/init
[    0.309187]     systemd.log_target=journal-or-kmsg
[    0.309188]     systemd.log_color=0
[    0.309190]     systemd.show_status=true
[    0.309192]     SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST=25
[    0.309193]   with environment:
[    0.309195]     HOME=/
[    0.309197]     TERM=linux
[    0.309198]     SYSTEMD_CGROUP_ENABLE_LEGACY_FORCE=1
[    0.309200]     BOOT_IMAGE=(hd0,gpt3)/vmlinuz
[    0.309202]     selinux=1
[    0.309203]     enforcing=1
```
```
bash-5.1# cat /proc/1/environ
HOME=/TERM=linuxSYSTEMD_CGROUP_ENABLE_LEGACY_FORCE=1BOOT_IMAGE=(hd0,gpt3)/vmlinuzselinux=1enforcing=1
```
And the output from the logging patch:
```
Sep 03 17:39:59 localhost systemd[1]: Mount rate limit burst configured to 5
Sep 03 17:39:59 localhost systemd[1]: Starting Generate network units from Kernel command line...
Sep 03 17:39:59 localhost systemd[1]: Finished Generate network units from Kernel command line.
Sep 03 17:39:59 localhost systemd[1]: Event source 0x55ca2e90cd50 (mount-monitor-dispatch) entered rate limit state.
Sep 03 17:39:59 localhost systemd[1]: Mount rate limit expired: 5 events exceeded in 1000000μs interval
Sep 03 17:40:00 localhost systemd[1]: Event source 0x55ca2e90cd50 (mount-monitor-dispatch) entered rate limit state.
Sep 03 17:40:00 localhost systemd[1]: Mount rate limit expired: 5 events exceeded in 1000000μs interval
Sep 03 17:40:01 localhost systemd[1]: Event source 0x55ca2e90cd50 (mount-monitor-dispatch) entered rate limit state.
Sep 03 17:40:01 localhost systemd[1]: Starting Generate network configuration...
Sep 03 17:40:01 localhost systemd[1]: Finished Generate network configuration.
Sep 03 17:40:01 localhost systemd[1]: Mount rate limit expired: 5 events exceeded in 1000000μs interval
Sep 03 17:40:02 localhost systemd[1]: Starting Generate additional settings for Kubernetes...
Sep 03 17:40:03 localhost systemd[1]: Finished Generate additional settings for Kubernetes.
```

bcressey commented on packages/systemd-257/9011-remove-cgroup-kernel-cmdline-requirement.patch:
```diff
@@ -0,0 +1,23 @@
+         if (!cg_is_legacy_enabled())
+                 return false;
+ 
+-        r = proc_cmdline_get_bool("SYSTEMD_CGROUP_ENABLE_LEGACY_FORCE", /* flags = */ 0, &b);
```
Since we're using bootconfig to deploy the `SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST` change, we could also use it for `SYSTEMD_CGROUP_ENABLE_LEGACY_FORCE` and then we won't need this patch.

vigh-m replied:
Yup! I have that change coming in the latest push

vigh-m commented:
⬆️ Changed the commit messages to be more clear about what was done and why. 
Also reworked the patch to remove `SYSTEMD_CGROUP_ENABLE_LEGACY_FORCE` to just be a bootconfig in the `release` package instead

bcressey commented on packages/release/release.spec:
```diff
@@ -322,10 +324,14 @@ install -p -m 0644 %{S:1400} %{buildroot}%{_cross_datadir}/logdog.d
 install -d %{buildroot}%{_cross_bootconfigdir}
 install -p -m 0644 %{S:1500} %{buildroot}%{_cross_bootconfigdir}/10-fips.conf
+install -p -m 0644 %{S:1501} %{buildroot}%{_cross_bootconfigdir}/10-mount-rate-limit-burst.conf
+install -p -m 0644 %{S:1502} %{buildroot}%{_cross_bootconfigdir}/10-cgroup-enable-legacy-force.conf
```
nit: I'd number these 11 and 12 to make the order more obvious, but it doesn't matter much in this case.

vigh-m replied:
Since I'm moving them to `systemd-257`,  I'll renumber them to 20 and 21 to separate them from the configs in `release`

bcressey commented on packages/release/release.spec:
```diff
@@ -322,10 +324,14 @@ install -p -m 0644 %{S:1400} %{buildroot}%{_cross_datadir}/logdog.d
 
 %files
+%{_cross_bootconfigdir}/10-mount-rate-limit-burst.conf
+%{_cross_bootconfigdir}/10-cgroup-enable-legacy-force.conf
```
I think these files should go in the `systemd-257` package - otherwise we're introducing new behavior to the existing variants on systemd 252.

vigh-m replied:
Yeah. I figured they would be ignored by `systemd-252` but I can move them to the `systemd-257` package

vigh-m committed:
systemd: migrate mount-rate patch to bootconfig

Move 9007-core-mount-increase-mount-rate-limit-burst-to-25.patch
to use the `SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST` env
variable instead. Using bootconfig allows us to set this as a default
fot all variants without changing the variant definitions.

Signed-off-by: Vighnesh Maheshwari <vighmah@amazon.com>

vigh-m committed:
systemd: remove systemd kernel cmdline req for cgroupsv1

Add a bootconfig to remove the requirement for
`SYSTEMD_CGROUP_ENABLE_LEGACY_FORCE` to be set on the kernel command line
to enable cgroupsv1. This maintains compatibility with the behaviour
in systemd-252

Signed-off-by: Vighnesh Maheshwari <vighmah@amazon.com>

vigh-m commented:
⬆️ Pushed all changes to the `systemd-257` package. Testing shows no change in behaviour
